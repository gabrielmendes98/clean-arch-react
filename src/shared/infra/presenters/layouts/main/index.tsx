import { Outlet } from 'react-router-dom';
import { Footer } from '../../footer';
import { Header } from '../../header';
import styles from './styles.module.scss';

export const MainLayout = () => (
  <div className={styles.container}>
    <Header />
    <main className={styles.main}>
      <Outlet />
    </main>
    <Footer />
  </div>
);
