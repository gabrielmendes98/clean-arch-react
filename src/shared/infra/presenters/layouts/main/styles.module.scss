@import 'shared/infra/styles/theme.scss';

.container {
  display: flex;
  flex-direction: column;
  height: 100%;
}

.main {
  flex: 1;
  background-color: $background-color;
}
