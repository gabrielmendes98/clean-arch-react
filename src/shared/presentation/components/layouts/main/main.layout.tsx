import { Outlet } from 'react-router-dom';
import { Footer } from '../../footer/footer.component';
import { Header } from '../../header/header.component';
import styles from './styles.module.scss';

export const MainLayout = () => (
  <div className={styles.container}>
    <Header />
    <main className={styles.main}>
      <Outlet />
    </main>
    <Footer />
  </div>
);
