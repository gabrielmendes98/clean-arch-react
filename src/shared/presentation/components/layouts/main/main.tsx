import { Outlet } from 'react-router-dom';
import { Footer } from '../../footer';
import { Header } from '../../header';
import style from './main.module.scss';

export const MainLayout = () => (
  <div className={style.container}>
    <Header />
    <main className={style.main}>
      <Outlet />
    </main>
    <Footer />
  </div>
);
