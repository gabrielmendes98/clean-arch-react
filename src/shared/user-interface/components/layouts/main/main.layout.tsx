import { Outlet } from 'react-router-dom';
import { Footer } from '../../footer/footer.component';
import { Header } from '../../header/header.component';
import styles from './main.module.scss';

export const MainLayout = () => (
  <div className={styles.box}>
    <Header />
    <main className={styles.main}>
      <Outlet />
    </main>
    <Footer />
  </div>
);
