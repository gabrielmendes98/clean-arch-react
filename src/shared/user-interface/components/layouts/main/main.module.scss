@import 'shared/user-interface/styles/theme.scss';

.box {
  display: flex;
  flex-direction: column;
  height: 100%;
}

.main {
  flex: 1;
  background-color: $background-color;
  padding: 50px;
}
