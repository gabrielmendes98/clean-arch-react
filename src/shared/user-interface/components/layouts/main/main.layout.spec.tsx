import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { render } from 'shared/testing/test-utils';
import { MainLayout } from './main.layout';

const Component = () => <div>some children</div>;

describe('MainLayout', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Routes>
          <Route element={<MainLayout />} path="/">
            <Route path="" element={<Component />} />
          </Route>
        </Routes>
      </BrowserRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
