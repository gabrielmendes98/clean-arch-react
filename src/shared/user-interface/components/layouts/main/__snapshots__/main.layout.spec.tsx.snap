// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="box"
  >
    <header
      class="header"
    >
      <nav>
        <ul>
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/login"
            >
              Login
            </a>
          </li>
          <li>
            <a
              href="/sign-up"
            >
              Registrar-se
            </a>
          </li>
          <li>
            <a
              href="/register-employees"
            >
              Cadastrar Funcionario
            </a>
          </li>
          <li>
            <a
              href="/employees"
            >
              Lista de Funcionarios
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <main
      class="main"
    >
      <div>
        some children
      </div>
    </main>
    <footer
      class="footer"
    >
      Â© Copyright 2022, Gabriel Santiago
    </footer>
  </div>
</DocumentFragment>
`;
